shared/src/main/scala/drt/shared/Api.scala

```diff
@@ -487,23 +545,18 @@ object FlightsApi {
 
       val newPortState: PortState = applyToPortState(updatedFlights, portState)
       val newDiff: PortStateDiff = portStateDiff(updatedFlights)
+
       (newPortState, newDiff)
     }
 
     def portStateDiff(updatedFlights: Seq[ApiFlightWithSplits]): PortStateDiff = {
       val removals = arrivalsToRemove.map(f => RemoveFlight(UniqueArrival(f)))
-      val newDiff = PortStateDiff(removals, updatedFlights, Seq(), Seq())
-      newDiff
+      PortStateDiff(removals, updatedFlights, Seq(), Seq())
     }
 
     def applyToPortState(updatedFlights: Seq[ApiFlightWithSplits], portState: PortState): PortState = {
-      val newFlights = updatedFlights.foldLeft(portState.flights) {
-        case (soFar, fws) => soFar.updated(fws.apiFlight.uniqueId, fws)
-      }
-      val newFlightsMinusRemovals = arrivalsToRemove.foldLeft(newFlights) {
-        case (minusRemovals, toRemove) => minusRemovals - toRemove.uniqueId
-      }
-      portState.copy(flights = newFlightsMinusRemovals)
+      val newFlights = portState.flights -- arrivalsToRemove.map(_.uniqueId) ++ updatedFlights.map(f => (f.apiFlight.uniqueId, f))
```

Comment:
confusing order of precedence here again.

Reply:
added brackets and extracted the map to `idsToRemove`